File: euler/02/2.cpp
#include <cstdio>
#include <cmath>

int main(){

	const int upper = 4*pow(10,6);

	int sum=0, f1=1, f2=2;

	while(f2<upper){
		if(f2 % 2 == 0){
			sum += f2;
		}
		f2 = f1 + f2;
		f1 = f2 - f1;
	}
	printf("The sum is %d\n", sum);

}

File: euler/02/2.pl

use strict;
use warnings;
use diagnostics;

my $upper = $ARGV[0];
if( not $upper ){
    $upper = 4000000;
}

my $sum =0;

my $n1 = 1;
my $n2 = 2;
while( $n2 < $upper ){
    if( $n2 % 2 == 0){
        $sum += $n2;
    }
    $n2 = $n1 + $n2;
    $n1 = $n2 - $n1;

}

print "The sum of all fibonacci numbers less than $upper is $sum\n"
